pvlib/temperature.py

```diff
@@ -821,3 +825,101 @@ def noct_sam(poa_global, temp_air, wind_speed, noct, module_efficiency,
+
+    The Prilliman et al. model [1]_ applies a weighted moving average to
+    the output of a steady-state cell temperature model to account for
+    a module's thermal inertia by smoothing the cell temperature's
+    response to changing weather conditions.
+
+    .. warning::
+        This implementation requires the time series inputs to be regularly
+        sampled in time with frequency less than 20 minutes.  Data with
+        irregular time steps should be resampled prior to using this function.
+
+    Parameters
+    ----------
+    temp_cell : pandas Series with DatetimeIndex
+        Cell temperature modeled with steady-state assumptions. [C]
+
+    wind_speed : pandas Series
+        Wind speed, adjusted to correspond to array height [m/s]
+
+    unit_mass : float, default 11.1
+        Total mass of module divided by its one-sided surface area [kg/m^2]
+
+    coefficients : 4-element list-like, optional
+        Values for coefficients a_0 through a_3, see Eq. 9 of [1]_
+
+    Returns
+    -------
+    temp_cell : pandas Series
+        Smoothed version of the input cell temperature [C]
+
+    Notes
+    -----
+    This smoothing model was developed and validated using the SAPM
+    cell temperature model for the steady-state input.
+
+    References
+    ----------
+    .. [1] M. Prilliman, J. S. Stein, D. Riley and G. Tamizhmani,
+       "Transient Weighted Moving-Average Model of Photovoltaic Module
+       Back-Surface Temperature," IEEE Journal of Photovoltaics, 2020.
+       :doi:`10.1109/JPHOTOV.2020.2992351`
+    """
+
+    time_step, window = _get_sample_intervals(temp_cell.index, 20)
+
+    if time_step >= 20:
+        warnings.warn("temperature.prilliman only applies smoothing when "
+                      "the sampling interval is shorter than 20 minutes "
+                      f"(input sampling interval: {time_step} minutes)")
+        # too coarsely sampled for smoothing to be relevant
+        return temp_cell
+
+    window = min(window,          # time series > 20 minutes total
+                 len(temp_cell))  # time series < 20 minutes total
+
+    # prefix with NaNs so that the rolling window is "full",
+    # even for the first actual value:
+    prefix = np.full(window, np.nan)
+    temp_cell_prefixed = np.append(prefix, temp_cell.values)
+
+    # get one row per 20-minute window
+    H = scipy.linalg.hankel(np.arange(window),
+                            np.arange(window - 1, len(temp_cell_prefixed)))
+    subsets = temp_cell_prefixed[H].T
+
+    # calculate weights for the values in each window
+    if coefficients is not None:
+        a = coefficients
+    else:
+        # values from [1], Table II
+        a = [0.0046, 0.00046, -0.00023, -1.6e-5]
+
+    wind_speed = wind_speed.values
+    P = a[0] + a[1]*wind_speed + a[2]*unit_mass + a[3]*wind_speed*unit_mass
+    timedeltas = np.arange(window, 0, -1) * (time_step*60)  # min to s
+    weights = np.exp(-P[:, np.newaxis] * timedeltas)
+
```

Comment:
The array manipulation below is very clever, but will fly over most people's heads (well, at least I had to look up the function examples). I don't think the code needs to change, and I don't have a great idea for how to make it more understandable, and it requires understanding the hankel matrix too, nor do I think it's your job to teach numpy here, but any further comments below might help. So yeah, tough spot, sorry.